scene/theme/default_theme.cpp

```diff
@@ -925,6 +925,37 @@ void fill_default_theme(Ref<Theme> &theme, const Ref<Font> &default_font, const
 	theme->set_constant("icon_max_width", "TabBar", 0);
 	theme->set_constant("outline_size", "TabBar", 0);
 
+	// TabBarBottom
+	Ref<StyleBoxFlat> style_tab_bottom_selected = make_flat_stylebox(style_normal_color, 10, 4, 10, 4, 0);
+	style_tab_bottom_selected->set_border_width(SIDE_BOTTOM, Math::round(2 * scale));
+	style_tab_bottom_selected->set_border_color(style_focus_color);
+	Ref<StyleBoxFlat> style_tab_bottom_unselected = make_flat_stylebox(style_pressed_color, 10, 4, 10, 4, 0);
+	// Add some spacing between unselected tabs to make them easier to distinguish from each other.
+	style_tab_bottom_unselected->set_border_width(SIDE_LEFT, Math::round(scale));
+	style_tab_bottom_unselected->set_border_width(SIDE_RIGHT, Math::round(scale));
+	style_tab_bottom_unselected->set_border_color(style_popup_border_color);
+	Ref<StyleBoxFlat> style_tab_bottom_disabled = style_tab_bottom_unselected->duplicate();
+	style_tab_bottom_disabled->set_bg_color(style_disabled_color);
+	Ref<StyleBoxFlat> style_tab_bottom_hovered = style_tab_bottom_unselected->duplicate();
+	style_tab_bottom_hovered->set_bg_color(Color(0.1, 0.1, 0.1, 0.3));
+	Ref<StyleBoxFlat> style_tab_bottom_focus = focus->duplicate();
+
+	theme->set_type_variation("TabBarBottom", "TabBar");
+
+	theme->set_stylebox("tab_selected", "TabBarBottom", style_tab_bottom_selected);
+	theme->set_stylebox("tab_hovered", "TabBarBottom", style_tab_bottom_hovered);
+	theme->set_stylebox("tab_unselected", "TabBarBottom", style_tab_bottom_unselected);
+	theme->set_stylebox("tab_disabled", "TabBarBottom", style_tab_bottom_disabled);
+	theme->set_stylebox("tab_focus", "TabBarBottom", style_tab_bottom_focus);
+
+	theme->set_type_variation("TabBarBottom", "TabContainer");
+
+	theme->set_stylebox("tab_selected", "TabBarBottom", style_tab_bottom_selected);
+	theme->set_stylebox("tab_hovered", "TabBarBottom", style_tab_bottom_hovered);
+	theme->set_stylebox("tab_unselected", "TabBarBottom", style_tab_bottom_unselected);
+	theme->set_stylebox("tab_disabled", "TabBarBottom", style_tab_bottom_disabled);
+	theme->set_stylebox("tab_focus", "TabBarBottom", style_tab_bottom_focus);
```

Comment:
Those should be inside the `TabContainer` theme (with each style ending with "_bottom"), since it's there they are being used, and to allow editing them in the editor.

Reply:
I feel like type variations are a good fit for bottom tabs, since the tabs themselves shouldn't care about being on the bottom. It also simplifies the code and simplifies configuring styles for tabs. The TabBarBottom variation can also be configured separately in the theme editor.
If its about letting the user configure the variation, I could only set the variation if it hasn't been set, like:
```
if (tabs_position == POSITION_TOP) {
	if (get_theme_type_variation() == "TabBarBottom") {
		set_theme_type_variation("");
	}
} else {
	if (get_theme_type_variation() == "") {
		set_theme_type_variation("TabBarBottom");
	}
}
```
Then if the user wants to change the tab bar position, the theme variation is in their control.

Reply:
The problem with this is that those styles will be hidden from the user, no way to know it via documentation, and no access via the inspector or the theme list. They would need to discover them from somewhere else somehow, and the only way to change them would be by creating a theme or doing it from code.

And besides all that, no other node does user-facing theming is this manner.

Reply:
We can improve the TabBar and TabContainer docs to mention when the theme variation should be used, like:
`tab_unselected`: "The style of the other, unselected tabs. If you need tabs at the [Constant POSITION_BOTTOM] and the [Constant POSITION_TOP], use `TabBarBottom` theme type variation to distinguish."

discoverability: 
when position changed to bottom in inspector: 
![image](https://github.com/godotengine/godot/assets/10054226/98fda89f-28c9-4cbc-939a-1b81745d5f4d)
![image](https://github.com/godotengine/godot/assets/10054226/2fe88043-d637-44af-8f50-63c5640a06a5)
in the theme editor when adding a type:
![image](https://github.com/godotengine/godot/assets/10054226/b6c017fe-54e7-47f5-9c71-b5242f9ecd3f)

As far as I know the only way to change the style without creating a theme or doing it from code is the style overrides, which works as expected.

Because of how variations work, when the regular TabContainer stylebox is changed and there is no TabBarBottom variation set, the style bottom tabs is set, so its not necessary to use the variation. 

Its not the same but, FileDialog and GraphEdit change some of their internal element's theme type variation, so if a user wants to change those parts they can use the type variations, and it isn't documented.

Reply:
> We can improve the TabBar and TabContainer docs to mention when the theme variation should be used

And what about being able to find those styles by simply searching for their names, like the theme items of other UI nodes?

> when position changed to bottom in inspector:

The way it's done here would actually create another problem on its own, what if the user is already using a theme override in their `TabContainer`? This would erase the value set by them without them knowing.

> Its not the same but, FileDialog and GraphEdit change some of their internal element's theme type variation

From what I saw, their variations are just empty styleboxes and small font tweaks. But even so, this shouldn't excuse going down that path in an even larger manner.

Reply:
> And what about being able to find those styles by simply searching for their names, like the theme items of other UI nodes?

The normal tab container tab styles (`tab_unselected`, etc.) are used to find the theme items when searching their names. If the user specifically wants to differentiate tabs at the bottom in a theme, the documentation for each of the theme items mentions the variation.
If the user is just trying to configure a tab container with tabs at the bottom, the normal theme items work fine. Changing between top and bottom tabs doesn't happen frequently enough that the theme items need to be separate.

Since tabs can't be at top and at bottom at the same time, having a theme variant is more intuitive than having extra theme items just for bottom tabs.

> what if the user is already using a theme override in their TabContainer?

The theme type variation does not get overridden if the user changed the value.

> From what I saw, their variations are just empty styleboxes and small font tweaks. But even so, this shouldn't excuse going down that path in an even larger manner.

I feel like this use case is very well suited to using theme type variations and should be allowed even if it sets a precedent.
Also, otherwise we would need to add a `use_bottom_style` flag to TabBar that only affects style and check it each time we draw a tab.